### PR TITLE
graph_scene: correct placement of labels

### PR DESCRIPTION
**Pull request by PhilippImhof**
Currently, when users use `add_end` or `add_start` in order to lengthen the axes a bit before/after the last tick, the labels may be misplaced, especially when using arrow tips. This PR fixes the problem.

## List of Changes
- For the x axis, do not use axis tip or ticks to determine where to put the label. Instead, use `get_corner`.
- ~~For the y axis, specify `aligned_edge` when using `next_to`, i.e. if for any reason someone wants the label to be below the y axis, it should be aligned w.r.t. its top edge.~~
- Remove `shift_onto_screen` from label placement. The label should go where the user asked to put it. If it does not entirely fit on the screen, they are free to chose another position.

## Motivation

I just don't like misplaced axis labels :) 

## Testing Status
Tested locally with several combinations.

## Acknowledgement
- [X] I have read the [Contributing Guidelines](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP))

### CODE REVIEW

**Comment by PhilippImhof** [comment]
My bad, I forgot to recreate the reference data for the test. Will do so ASAP
Figure_1

**Review by leotrs — COMMENTED** [review]
This LGTM. I know you fixed an old test, but would you be able to add a new one so we don't regress on this? One that directly calls add_end() for example

**Comment by PhilippImhof** [comment]
Not quite sure I understand. You mean add a test that makes use of the `add_start` and/or `add_end` functionality?

**Comment by leotrs** [comment]
Yes

**Comment by PhilippImhof** [comment]
Done. I came across some strange behavior (not related to my changes though) that I will have to look at more in depth. Basically, the label is sometimes put at the wrong place, e.g. when there is "not enough space" above the y axis. I will open a separate issue for that, providing more details.

File: tests/test_graphical_units/test_axes.py [hunk 1]
```diff
@@ -3,7 +3,9 @@
 from manim import *
 from ..utils.testing_utils import get_scenes_to_test
 from ..utils.GraphicalUnitTester import GraphicalUnitTester
-#from ..helpers.graphical_units import set_test_scene
+
+# from ..helpers.graphical_units import set_test_scene
+
```
Comment (leotrs) [inline tests/test_graphical_units/test_axes.py:8]:
```suggestion
```

File: tests/test_graphical_units/test_axes.py [hunk 2]
```diff
@@ -41,4 +43,5 @@ def construct(self):
 def test_scene(scene_to_test, tmpdir, show_diff):
     GraphicalUnitTester(scene_to_test[1], MODULE_NAME, tmpdir).test(show_diff=show_diff)
 
-#set_test_scene(AxesTest, "graph")
+
+# set_test_scene(AxesTest, "graph")
```
Comment (leotrs) [inline tests/test_graphical_units/test_axes.py:47]:
```suggestion
```

**Review by leotrs — CHANGES_REQUESTED** [review]

**Review by leotrs — COMMENTED** [review]
I see you've added or modified AxesTest and PlotFunctions, but I don't see which one of them actually calls the functions add_start or add_end? :thinking: Am I missing anything?

**Comment by PhilippImhof** [comment]
I'd say `tests/test_graphical_units/test_axes.py` has that on L15-16 and L20-21. :)

**Comment by PhilippImhof** [comment]
After having digged deeper, I suggest to fix #663 directly with this PR as soon as we have a consensus on what behavior to adopt.

**Review by leotrs — APPROVED** [review]

**Review by cobordism — APPROVED** [review]
looks fine to me.

**Pull request merged**